swig/python/CMakeLists.txt

```diff
@@ -413,8 +413,8 @@ if(FLAMEGPU_BUILD_PYTHON_VENV)
     # Install the wheel from the dist dir. Specify a version incase the same build dir has been used for a previous build
     # Relaxed the no-index option to allow setup.py to install requirements
     COMMAND ${VENV_PIP} install --force-reinstall --find-links=dist -U ${PYTHON_DISTRIBUTION_NAME}==${FLAMEGPU_VERSION_PYTHON}
-    # install pytest
-    COMMAND ${VENV_PIP} install pytest
+    # install pytest and astpretty 
+    COMMAND ${VENV_PIP} install pytest astpretty
```

Comment:
Is there a reason to remove it from the venv, if it's used by the codegen tests?

Reply:
this is adding it to the venv if pytest is also being installed. 

Previously it would be implicitly installed as a dependency of pyflamegpu.

The changes to test_codegen.py itself making it optional within the test is not really neccesarry, just adds a bit more flexibility, it makes it so that just pyflamegpu and pytest are required, but if codegen debugging is wanted then astpretty might help.